iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -166,15 +163,20 @@ LogicalResult TileSizeCalculator::inferTileAndWorkgroupSize(
     workgroupSize = {tileSizeX, tileSizeY, 1};
     return success();
   }
-  if (opInfo & OpInfo::Matmul) {
-    // TODO: For now just hard wire this, but we can do better.
-    tileSizes = {8, 8, 4};
-    workgroupSize = {8, 8, 1};
-    return success();
-  }
-  if (opInfo & OpInfo::BatchMatmul) {
-    tileSizes = {2, 8, 8, 4};
-    workgroupSize = {8, 8, 2};
+  if ((opInfo & OpInfo::Matmul) || (opInfo & OpInfo::BatchMatmul)) {
+    // Set tile size {M, N, K} = {maxWorkgroupSize /
+    // ceil(sqrt(maxWorkgroupSize)), ceil(sqrt(maxWorkgroupSize)), 32}
+    int64_t tileSizeN = std::max<int64_t>(
+        1, static_cast<int64_t>(llvm::PowerOf2Ceil(
+               static_cast<uint64_t>(std::sqrt(maxWorkgroupSize)))));
+    int64_t tileSizeM = maxWorkgroupSize / tileSizeN;
```

Comment:
According to the comment, I'm expecting

tileSizeN = std::ceil(std::sqrt(maxWorkgropuSize));